### ISSUE TITLE
RLVMR test set performance is higher than the training set

### ISSUE THREAD

**Issue by tianleli1** [Tencent/digitalhuman#21]
Thank you for your work! I'm reproducing RLVMR-qwen2.5-1.5B-instruct-alfworld-L0, using a 200-sample cold start and 100 steps of reinforcement learning. Here are my results: episode/success_rate: 0.906, 'val/success_rate': np.float64(0.9296875). Is it normal for the test set to outperform the training set? Is there a possibility of data leakage (during the cold start)? Also, is there a separate script for testing the model? Thanks for your time.

**Comment by congli01** [comment]
I also have a similar question. I found that simply using a cold start process can improve the val/SR.

I used the expert data provided in the repo to prompt GPT-4o to generate reasoning, constructing a cold start dataset containing about 200 traj for SFT, and then trained qwen2.5-1.5B-instruct using GRPO. The `val/SR` reached 0.92118 at 75 step. 

<img width="1235" height="723" alt="Image" src="https://github.com/user-attachments/assets/f82e5bbe-f8d7-4ea8-ba9d-ada3f73d89bd" />

**Comment by hrwise-nlp** [comment]
Maybe it is important to check the SR after SFT stage.

**Comment by zzjweb** [comment]
Hi, we compared multiple framework implementations and found that this behavior is possible for the "L0" split. According to the original authors, both the Seen test and training sets share known task instances. If you only use the first few hundred expert trajectories, there should be no risk of data leakage.  

To fully evaluate generalization, you may set `generalization_level=2`(L2). For testing only, add `+trainer.val_only=True` to your launch script.
